### PR TITLE
Fix QA fail #1319 allow users to change the target language of their translation

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Fix of QA fail #1319 allow users to change the target language of their translation

Changes in this pull request:
- Added handling of merge on target language change to existing project. 
- Added new task for project merge

### CODE REVIEW

File: app/src/main/java/com/door43/translationstudio/newui/newtranslation/NewTargetTranslationActivity.java [hunk 1]
```diff
@@ -345,4 +444,54 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
+            setResult(results, data);
+            finish();
+        }
+    }
+
+
+    /**
+     * for keeping track if dialog is being shown for orientation changes
+     */
+    public enum eDialogShown {
```
Comment (neutrinog) [inline app/src/main/java/com/door43/translationstudio/newui/newtranslation/NewTargetTranslationActivity.java]: Don't begin an object definition with a lower case and don't prepend the type to the name. This should just be `DialogShown`.

Comment (neutrinog) [inline app/src/main/java/com/door43/translationstudio/newui/newtranslation/NewTargetTranslationActivity.java]: Also, unless you need to associate a string value to the enum there is no need for this boiler plate code.

The following should be sufficient.
```
public enum DialogShown {
  NONE,
  RENAME_CONFICT
}
```

File: app/src/main/java/com/door43/translationstudio/tasks/MergeTargetTranslationTask.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+import com.door43.translationstudio.App;
+import com.door43.translationstudio.core.TargetTranslation;
+import org.unfoldingword.tools.taskmanager.ManagedTask;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Pulls down changes from a remote target translation repository
```
Comment (neutrinog) [inline app/src/main/java/com/door43/translationstudio/tasks/MergeTargetTranslationTask.java]: This is an inaccurate description of what this class does.

Comment (PhotoNomad0) [inline app/src/main/java/com/door43/translationstudio/tasks/MergeTargetTranslationTask.java]: Oops

File: app/src/main/java/com/door43/translationstudio/newui/newtranslation/NewTargetTranslationActivity.java (hunk 1 above)
Comment (PhotoNomad0) [inline app/src/main/java/com/door43/translationstudio/newui/newtranslation/NewTargetTranslationActivity.java]: @neutrinog - It's not about strings here, but converting back and forth to integers in order to save state.  Is there an easier way?  Java doesn't allow casting between enums and integers.

**Comment by PhotoNomad0** [comment]
@neutrinog - ready for review.

**Comment by PhotoNomad0** [comment]
@neutrinog - OK got things merged.  After doing some reading on enums, came up with a much simpler enum definition that can handle conversion to/from int.  It also maintains encapsulation by wrapping the details of working with ordinals.

    public enum DialogShown {
        NONE,
        RENAME_CONFLICT;
        
        public int getValue() {
            return this.ordinal();
        }

        public static DialogShown fromInt(int ordinal, DialogShown defaultValue) {
            if (ordinal > 0 && ordinal < DialogShown.values().length) {
                return DialogShown.values()[ordinal];
            }
            return defaultValue;
        }
    }

**Pull request merged**